packages/cli/src/cli/serve.rs

```diff
@@ -41,11 +42,28 @@ impl Serve {
 
         crate_config.set_cargo_args(self.serve.cargo_args);
 
-        let platform = self
+        let mut platform = self
             .serve
-            .platform
-            .unwrap_or(crate_config.dioxus_config.application.default_platform);
+            .platform;
+
+        if platform.is_none() {
+            if let Some(dependency) = &crate_config.manifest.dependencies.get("dioxus") {
+                let features = match dependency {
+                    Inherited(detail) => detail.features.to_vec(),
+                    Detailed(detail) => detail.features.to_vec(),
+                    Simple(_) => vec![]
+                };
+
+                platform = features
+                    .iter()
+                    .next()
+                    .and_then(|first_feature| serde_json::from_str(&format!(r#""{}""#, first_feature)).ok());
```

Comment:
I think we should choose the first feature that matches a valid platform:

```
fullstack
desktop
mobile
web
ssr
liveview
axum
tui
```

Either that or complain if we see two platforms enabled at once.

Otherwise, fantastic work, thank you!

Reply:
Oh, I only found the Platform-Enum in cli-config containing
* web
* desktop
* fullstack
Where are the others located?

Currently the first entry is always picked, I will adjust it to search for the first valid platform.

Reply:
Oh, we might only support those platforms in the CLI then. That list is coming from the valid features on `dioxus/packages/cargo.toml`.